### PR TITLE
Contexts & contextual data

### PR DESCRIPTION
**Pull request by kashike**
Supersedes https://github.com/SpongePowered/SpongeCommon/pull/304
[API](https://github.com/SpongePowered/SpongeAPI/pull/1046) | [**Common**](https://github.com/SpongePowered/SpongeCommon/pull/451) | [Forge](https://github.com/SpongePowered/SpongeForge/pull/522)

(and yes, this intentionally has just the submodule update currently)

### CODE REVIEW

File: src/main/java/org/spongepowered/common/data/processor/context/common/AbstractContextAwareMultiDataSingleTargetProcessor.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import org.spongepowered.common.SpongeImpl;
+import org.spongepowered.common.data.context.ContextAwareDataProcessor;
+import org.spongepowered.common.data.processor.common.AbstractMultiDataSingleTargetProcessor;
+
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public abstract class AbstractContextAwareMultiDataSingleTargetProcessor<Container, Context extends DataContext, M extends DataManipulator<M, I>, I extends
```
Comment (Zarthus) [inline src/main/java/org/spongepowered/common/data/processor/context/common/AbstractContextAwareMultiDataSingleTargetProcessor.java:47]: I still think ChickenSoup would have been a better name for this class.

**Comment by ryantheleach** [comment]
People are constantly asking about features that would suit data contexts well. I know this is a very old PR, but is there any encountered issues with getting this up to date or mergable?

**Comment by ryantheleach** [comment]
This is currently stalled on the large amount of work needed + the current merge conflicts, + the Data API changing underneath this PR. Once the Data API is more stable, this PR will be updated and reviewed.

**Comment by Zidane** [comment]
Closing this given the age that has passed and the lack of activity. If someone wants to continue the work done here then we can re-open at a later time.

**Pull request closed without merging**